spot_wrapper/spot_graph_nav.py

```diff
@@ -0,0 +1,796 @@
+        response: map_processing_pb2.ProcessAnchoringResponse = (
+            self._map_processing_client.process_anchoring(
+                params=map_processing_pb2.ProcessAnchoringRequest.Params(),
+                modify_anchoring_on_server=True,
+                stream_intermediate_results=False,
+            )
+        )
+        if response.status == map_processing_pb2.ProcessAnchoringResponse.STATUS_OK:
+            self._logger.info(
+                "Optimized anchoring after {} iteration(s).".format(response.iteration)
+            )
+            return True, "Successfully optimized anchoring."
+        else:
+            self._logger.error("Error optimizing {}".format(response))
+            if (
+                response.status
+                == map_processing_pb2.ProcessAnchoringResponse.STATUS_MISSING_WAYPOINT_SNAPSHOTS
+            ):
+                return False, "Missing waypoint snapshots."
+            elif (
+                response.status
+                == map_processing_pb2.ProcessAnchoringResponse.STATUS_INVALID_GRAPH
+            ):
+                return False, "Invalid graph."
+            elif (
+                response.status
+                == map_processing_pb2.ProcessAnchoringResponse.STATUS_OPTIMIZATION_FAILURE
+            ):
+                return False, "Optimization failure."
+            elif (
+                response.status
+                == map_processing_pb2.ProcessAnchoringResponse.STATUS_INVALID_PARAMS
+            ):
+                return False, "Invalid parameters."
+            elif (
+                response.status
+                == map_processing_pb2.ProcessAnchoringResponse.STATUS_CONSTRAINT_VIOLATION
+            ):
+                return False, "Constraint violation."
+            elif (
+                response.status
+                == map_processing_pb2.ProcessAnchoringResponse.STATUS_MAX_ITERATIONS
+            ):
+                return False, "Max iterations, timeout."
+            elif (
+                response.status
+                == map_processing_pb2.ProcessAnchoringResponse.STATUS_MAX_TIME
+            ):
+                return False, "Max time reached, timeout."
+            elif (
+                response.status
+                == map_processing_pb2.ProcessAnchoringResponse.STATUS_INVALID_HINTS
+            ):
+                return False, "Invalid hints."
+            elif (
+                response.status
+                == map_processing_pb2.ProcessAnchoringResponse.STATUS_MAP_MODIFIED_DURING_PROCESSING
+            ):
+                return False, "Map modified during processing."
+            elif (
+                response.status
+                == map_processing_pb2.ProcessAnchoringResponse.STATUS_INVALID_GRAVITY_ALIGNMENT
+            ):
+                return False, "Invalid gravity alignment."
+            else:
+                return False, "Unknown error during anchoring optimization."
```

Comment:
```suggestion
            self._logger.error("Error optimizing {}".format(response))
            status_mapping = {
                map_processing_pb2.ProcessAnchoringResponse.STATUS_MISSING_WAYPOINT_SNAPSHOTS: "Missing waypoint snapshots.",
                map_processing_pb2.ProcessAnchoringResponse.STATUS_INVALID_GRAPH: "Invalid graph.",
                map_processing_pb2.ProcessAnchoringResponse.STATUS_OPTIMIZATION_FAILURE: "Optimization failure.",
                map_processing_pb2.ProcessAnchoringResponse.STATUS_INVALID_PARAMS: "Invalid parameters.",
                map_processing_pb2.ProcessAnchoringResponse.STATUS_CONSTRAINT_VIOLATION: "Constraint violation.",
                map_processing_pb2.ProcessAnchoringResponse.STATUS_MAX_ITERATIONS: "Max iterations, timeout.",
                map_processing_pb2.ProcessAnchoringResponse.STATUS_MAX_TIME: "Max time reached, timeout.",
                map_processing_pb2.ProcessAnchoringResponse.STATUS_INVALID_HINTS: "Invalid hints.",
                map_processing_pb2.ProcessAnchoringResponse.STATUS_MAP_MODIFIED_DURING_PROCESSING: "Map modified during processing.",
                map_processing_pb2.ProcessAnchoringResponse.STATUS_INVALID_GRAVITY_ALIGNMENT: "Invalid gravity alignment.",
            }
            if response.status in status_mapping:
                return False, status_mapping[response.status]

            return False, "Unknown error during anchoring optimization."
```

Another mapping to simplify if statements.

Reply:
Agreed, done in [392da89](https://github.com/bdaiinstitute/spot_wrapper/pull/6/commits/392da898d027fc716a17f1cd7ff487183cc192cb)